lib/matplotlib/ticker.py

```diff
@@ -532,8 +532,11 @@ def _format_maybe_minus_and_locale(self, fmt, arg):
         """
         Format *arg* with *fmt*, applying Unicode minus and locale if desired.
         """
-        return self.fix_minus(locale.format_string(fmt, (arg,), True)
-                              if self._useLocale else fmt % arg)
+        formatted = self.fix_minus(locale.format_string(fmt, (arg,), True)
+                                   if self._useLocale else fmt % arg)
+        if (self.get_useMathText()):  # removed unintended space after comma
+            formatted = formatted.replace(',', '{,}')
```

Comment:
This would incorrectly also escape commas actually present in `fmt`.

I guess a simple-ish solution could be to str.split `fmt` over commas, apply locale.format_string over each part, escape resulting commas over each formatted part, and join again using unescaped commas.